### PR TITLE
Added html content and calculation call to Deductions Summary

### PR DESCRIPTION
**Pull request by jameshforster**

### CODE REVIEW

File: app/views/calculation/resident/deductionsSummary.scala.html [hunk 1]
```diff
@@ -1,9 +1,66 @@
...
+@import uk.gov.hmrc.play.views.helpers.MoneyPounds
+
+@(gainAnswers: YourAnswersSummaryModel, deductionAnswers: ChargeableGainAnswers, result: ChargeableGainResultModel)(implicit request: Request[_])
 
 @resident_main_template(
-    title = Messages("calc.resident.summary.title"),
-    backLink = None,
-    articleLayout = false
+title = Messages("calc.resident.summary.title"),
+backLink = Some(controllers.resident.routes.GainController.improvements().toString),
```
Comment (DavidWoodward37) [inline app/views/calculation/resident/deductionsSummary.scala.html]: Unless you intended this to be fixed at a later date I think this needs the conditional back linking the same as the previous-taxable-gains page. I'd wait for mac's merge though.

File: test/connectors/CalculatorConnectorSpec.scala [hunk 2]
```diff
@@ -21,6 +21,8 @@ import java.util.UUID
 import common.KeystoreKeys
 import common.nonresident.CustomerTypeKeys
 import models.nonresident._
+import models.resident
+import models.resident.ChargeableGainAnswers
```
Comment (DavidWoodward37) [inline test/connectors/CalculatorConnectorSpec.scala]: models.resident._? should save you a few words everywhere.

File: test/views/resident/DeductionsSummaryViewSpec.scala [hunk 3]
```diff
@@ -17,14 +17,37 @@
...
+      None,
+      Some(LossesBroughtForwardModel(false)),
+      None,
+      None)
+    lazy val results = ChargeableGainResultModel(BigDecimal(50000),
+      BigDecimal(38900),
+      BigDecimal(11100),
+      BigDecimal(11100))
+    
+    lazy val view = views.html.calculation.resident.deductionsSummary(gainAnswers, deductionAnswers, results)(fakeRequestWithSession)
```
Comment (DavidWoodward37) [inline test/views/resident/DeductionsSummaryViewSpec.scala]: Could shorten the imports here to {resident => views}

File: test/views/resident/DeductionsSummaryViewSpec.scala [hunk 4]
```diff
@@ -34,6 +57,503 @@ class DeductionsSummaryViewSpec extends UnitSpec with WithFakeApplication with F
...
+        backLink.attr("id") shouldBe "back-link"
+      }
+
+      s"has the text '${commonMessages.calcBaseBack}'" in {
+        backLink.text shouldBe commonMessages.calcBaseBack
+      }
+
+      s"has a link to '${routes.GainController.improvements().toString()}'" in {
+        backLink.attr("href") shouldBe routes.GainController.improvements().toString
+      }
```
Comment (DavidWoodward37) [inline test/views/resident/DeductionsSummaryViewSpec.scala]: Same point as the comment above about back links.

File: test/connectors/CalculatorConnectorSpec.scala (hunk 2 above)
Comment (jameshforster) [inline test/connectors/CalculatorConnectorSpec.scala]: Cannot use, would result in clashes with non-resident models.

File: app/controllers/resident/SummaryController.scala [hunk 5]
```diff
@@ -29,6 +30,16 @@ trait SummaryController extends FeatureLock {
...
   val summary = FeatureLockForRTT.async { implicit request =>
+
+    def buildPreviousTaxableGainsBackUrl(chargeableGainAnswers: ChargeableGainAnswers)(implicit hc: HeaderCarrier): Future[String] = {
+      (chargeableGainAnswers.otherPropertiesModel.getOrElse(OtherPropertiesModel(false)).hasOtherProperties
+        , chargeableGainAnswers.broughtForwardModel.getOrElse(LossesBroughtForwardModel(false)).option) match {
+        case (true, _) => Future.successful(routes.DeductionsController.annualExemptAmount().url)
+        case (false, true) => Future.successful(routes.DeductionsController.lossesBroughtForwardValue().url)
+        case (false, false) => Future.successful(routes.DeductionsController.lossesBroughtForward().url)
+      }
+    }
```
Comment (DavidWoodward37) [inline app/controllers/resident/SummaryController.scala]: I think you are now missing tests in a Summary Action Spec for which URL a back link has based on the answers.

**Comment by jameshforster** [comment]
Missing tests have been added.

**Pull request merged**